### PR TITLE
Refactor/34411 lhn previews refactor

### PR DESCRIPTION
**Pull request by koko57**
### Details

PART OF THE REFACTOR: moved some logic to `getLastMessageTextForReport`, created a function to get last actor displayName

### Fixed Issues

$ https://github.com/Expensify/App/issues/34411
PROPOSAL:

### Tests

Archived Rooms
1. [User A] Create a new workspace (or use an existing one)
2. [User A] Invite another user to the workspace (User B)
3. [User A] Delete the workspace or remove User B from the workspace (ideally check both)
4. Observe the LHN previews for both User A and User B (you can compare it with staging)
5. Open Search Page - observe the previews. ([reference here](https://github.com/Expensify/App/issues/34411#issuecomment-1906612339))

Last Actor Display Name
1. For all the chat types the last actor name should be displayed (with the colon) verify that it's displayed properly.

- [x] Verify that no errors appear in the JS console

### QA Steps

Archived Rooms
1. [User A] Create a new workspace (or use an existing one)
2. [User A] Invite another user to the workspace (User B)
3. [User A] Delete the workspace or remove User B from the workspace (ideally check both)
4. Observe the LHN previews for both User A and User B (you can compare it with staging)
5. Open Search Page - observe the previews.

Last Actor Display Name
1. For all the chat types the last actor name should be displayed (with the colon) verify that it's displayed properly.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot_1706018899
Screenshot_1706018920

</details>

<details>
<summary>iOS: Native</summary>

Simulator Screenshot - iPhone 15 Pro - 2024-01-23 at 14 57 24
Simulator Screenshot - iPhone 15 Pro - 2024-01-23 at 14 57 39

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Simulator Screenshot - iPhone 15 Pro - 2024-01-23 at 15 26 09
Simulator Screenshot - iPhone 15 Pro - 2024-01-23 at 15 26 24

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-01-23 at 12 47 29
Screenshot 2024-01-23 at 12 53 25

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-01-23 at 14 20 34
Screenshot 2024-01-23 at 14 21 28

</details>

### CODE REVIEW

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -394,12 +394,27 @@ function getAllReportErrors(report, reportActions) {
     return allReportErrors;
 }
 
+/**
+ * Get the last actor display name from last actor details.
+ * @param lastActorDetails
+ * @param hasMultipleParticipants
+ * @returns {String}
+ */
+
```
Comment (mountiny) [inline src/libs/OptionsListUtils.js:403]:
```suggestion
```

File: src/libs/OptionsListUtils.js [hunk 2]
```diff
@@ -526,24 +550,11 @@ function createOption(accountIDs, personalDetails, report, reportActions = {}, {
...
+        const lastActorDisplayName = getLastActorDisplayName(lastActorDetails, hasMultipleParticipants);
+        const lastMessageTextFromReport = getLastMessageTextForReport(report, lastActorDetails);
         let lastMessageText = lastMessageTextFromReport;
 
-        if (result.isArchivedRoom) {
-            const archiveReason =
-                (lastReportActions[report.reportID] && lastReportActions[report.reportID].originalMessage && lastReportActions[report.reportID].originalMessage.reason) ||
-                CONST.REPORT.ARCHIVE_REASON.DEFAULT;
-            lastMessageText = Localize.translate(preferredLocale, `reportArchiveReasons.${archiveReason}`, {
-                displayName: archiveReason.displayName || PersonalDetailsUtils.getDisplayNameOrDefault(lastActorDetails),
```
Comment (mountiny) [inline src/libs/OptionsListUtils.js:542]: This ` archiveReason.displayName` has been wrong?

File: src/libs/SidebarUtils.ts [hunk 3]
```diff
@@ -365,31 +349,12 @@ function getOptionData({
             : null;
     }
 
-    const shouldShowDisplayName = hasMultipleParticipants && lastActorDetails?.accountID && Number(lastActorDetails.accountID) !== currentUserAccountID;
-    const lastActorName = lastActorDetails?.firstName ?? lastActorDetails?.displayName;
-    const lastActorDisplayName = shouldShowDisplayName ? lastActorName : '';
+    const lastMessageTextFromReport = OptionsListUtils.getLastMessageTextForReport(report, lastActorDetails, policy);
+    const lastActorDisplayName = OptionsListUtils.getLastActorDisplayName(lastActorDetails, hasMultipleParticipants);
```
Comment (mountiny) [inline src/libs/SidebarUtils.ts:353]: Purely OCD reasons, to keep the same order as in the other usecase
```suggestion
    const lastActorDisplayName = OptionsListUtils.getLastActorDisplayName(lastActorDetails, hasMultipleParticipants);
    const lastMessageTextFromReport = OptionsListUtils.getLastMessageTextForReport(report, lastActorDetails, policy);
```

File: src/libs/OptionsListUtils.js [hunk 4]
```diff
@@ -436,6 +451,15 @@ function getLastMessageTextForReport(report) {
         lastMessageTextFromReport = lodashGet(lastReportAction, 'message[0].text', '');
     } else if (ReportActionUtils.isCreatedTaskReportAction(lastReportAction)) {
         lastMessageTextFromReport = TaskUtils.getTaskCreatedMessage(lastReportAction);
+    } else if (ReportUtils.isArchivedRoom(report)) {
+        const archiveReason =
+            (lastReportActions[report.reportID] && lastReportActions[report.reportID].originalMessage && lastReportActions[report.reportID].originalMessage.reason) ||
```
Comment (mountiny) [inline src/libs/OptionsListUtils.js:456]: Should we also make sure the .reason is one of the 3 options?

**Review by mountiny — COMMENTED** [review]
@koko57 I think its looking good, great job!

File: src/libs/OptionsListUtils.js (hunk 2 above)
Comment (koko57) [reply src/libs/OptionsListUtils.js:542]: it should be a string, shouldn't be? Or I'm missing a case where it can be an object?

**Comment by koko57** [comment]
@mountiny thanks! I will fill in the author's checklist shortly. I will also check what's wrong with the reassure tests, but I've noticed that this particular test is failing for other PRs as well

File: src/libs/OptionsListUtils.js (hunk 4 above)
Comment (koko57) [reply src/libs/OptionsListUtils.js:456]: You mean one of these 3: `CONST.REPORT.ARCHIVE_REASON.ACCOUNT_CLOSED`, `CONST.REPORT.ARCHIVE_REASON.REMOVED_FROM_POLICY`, `CONST.REPORT.ARCHIVE_REASON.POLICY_DELETED`? Like it was checked in [SidebarUtils](https://github.com/Expensify/App/blob/c0608977d90835b34382325732e755733c034240/src/libs/SidebarUtils.ts#L378-L391)? Can we get another reason (apart from these [ones](https://github.com/Expensify/App/blob/c0608977d90835b34382325732e755733c034240/src/languages/en.ts#L507-L517))? Actually, now I see that we were not showing the `MERGED` reason there - we would also need to pass the `oldDisplayName` if so. Shou,d it be displayed or should we display `DEFAULT` instead

**Comment by koko57** [comment]
@mountiny could you please take a look on these comments? 🙏🏻 
https://github.com/Expensify/App/pull/34872#discussion_r1463002686
https://github.com/Expensify/App/pull/34872#discussion_r1462940070

File: src/libs/OptionsListUtils.js (hunk 4 above)
Comment (mountiny) [reply src/libs/OptionsListUtils.js:456]: Lets focus on keeping the current behaviour in place. I think we can skip adding the merged consideration for now

File: src/libs/OptionsListUtils.js (hunk 2 above)
Comment (mountiny) [reply src/libs/OptionsListUtils.js:543]: Seems like you are right so it has been wrong 😄

**Comment by mountiny** [comment]
@koko57 answered

**Comment by koko57** [comment]
ok, @mountiny so it's ready to review. It also includes the fix @trjExpensify mentioned https://github.com/Expensify/App/issues/34411#issuecomment-1906612339

**Comment by mountiny** [comment]
@koko57 there seems to be some regression with archived workspace chat that had some requests in it

image

On staging we correctly show that its archived, on the build you can see there is the owes message

**Comment by koko57** [comment]
@mountiny ah yes, I didn't thought of that case - this logic for archived room was overwriting the last message no matter what the last action was. I will look into it if I can change the conditions (without breaking the other messages) and leave the logic in `getLastMessageTextForReport `

**Comment by koko57** [comment]
@mountiny fixed! I'm resolving now the conflicts after TS migration. I have one question: do we really need to check if last action was CLOSED like it was introduced here?
Screenshot 2024-01-26 at 09 30 16

**Comment by mountiny** [comment]
@koko57 good question, there should be no other actions when it was archived. I think its safer to check for it though

**Comment by koko57** [comment]
ok, now I see this check was added for TS to work

**Comment by eVoloshchak** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by eVoloshchak** [comment]
An owner's email is displayed instead of "This workspace chat is no longer active.." until you open a report 

https://github.com/Expensify/App/assets/9059945/e9121a0b-1129-49ae-ade8-1ccb3460324e

This happens only on DEV, [ad-hoc build](https://github.com/Expensify/App/pull/34872#issuecomment-1910163382) works fine
@mountiny, could you re-run the workflow to generate a new ad-hoc build please? Looks like this bug might be DEV only

**Comment by eVoloshchak** [comment]
Screenshot 2024-01-26 at 19 45 40
Screenshot 2024-01-26 at 19 45 28

Shouldn't we show `(archived)` label for iou requests in an archived workspace?

**Comment by mountiny** [comment]
@koko57 could you sync with main before I make another build

**Comment by mountiny** [comment]
https://github.com/Expensify/App/actions/runs/7698510629

**Comment by mountiny** [comment]
@eVoloshchak whats the update here?

**Comment by eVoloshchak** [comment]
@koko57, [this](https://github.com/Expensify/App/pull/34872#issuecomment-1912523526) behavior is still present in https://34872.pr-testing.expensify.com/


https://github.com/Expensify/App/assets/9059945/863cafb0-f3d3-41c3-9917-28698dce921c



It doesn't work properly on https://new.expensify.com/ either, but instead of an email it displays "This chat room has been archived"


https://github.com/Expensify/App/assets/9059945/1c39aea5-d6bf-4c83-89c2-288d5ca5b268

**Comment by mountiny** [comment]
I believe the difference is that in the ad hoc build we rely on the CLOSED report action only whereas in production it seems to know the report state is closed and does not show the message. I believe we might not be loading the Closed report actions in OpenApp and that has been discussed in the past, we probably will not change anything about it. 

I think if the report is the in closed state/status, we should keep the same behaviour you can see in production

**Comment by koko57** [comment]
@mountiny @eVoloshchak so I will revert these changes and leave the ones that work - getting last act

> @mountiny @eVoloshchak maybe if it will happen again I revert the changes for the archived rooms, leaving only the changes for displaying last actor's name and the fix for Workspace preview?

So I think it's better to revert these changes and I'll try to fix it in another PR, with other changes

**Comment by koko57** [comment]
@mountiny thanks for the hint! I was able to change the condition a little bit, so now we've got the same behaviour as in the production (We can think of fixing it later).

https://github.com/Expensify/App/assets/28020445/2211e773-4372-4a06-8f6f-21355fa5e2bf

**Comment by mountiny** [comment]
@eVoloshchak Can you please proceed with the checklist? Thanks!

**Review by eVoloshchak — APPROVED** [review]
LGTM!

**Review by mountiny — APPROVED** [review]
Thanks! getting better bit by bit

File: src/libs/OptionsListUtils.ts [hunk 5]
```diff
@@ -489,15 +489,43 @@ function getAllReportErrors(report: OnyxEntry<Report>, reportActions: OnyxEntry<
...
+}
+
 /**
  * Get the last message text from the report directly or from other sources for special cases.
  */
-function getLastMessageTextForReport(report: OnyxEntry<Report>): string {
+function getLastMessageTextForReport(report: OnyxEntry<Report>, lastActorDetails: Partial<PersonalDetails> | null, policy?: OnyxEntry<Policy>): string {
     const lastReportAction = allSortedReportActions[report?.reportID ?? '']?.find((reportAction) => ReportActionUtils.shouldReportActionBeVisibleAsLastAction(reportAction)) ?? null;
+    // some types of actions are filtered out for lastReportAction, in some cases we need to check the actual last action
+    const lastOriginalReportAction = lastReportActions[report?.reportID ?? ''] ?? null;
```
Comment (shubham1206agra) [inline src/libs/OptionsListUtils.ts:507]: You have missed to consider a case where the comment is flagged and this caused a bug. More details - https://github.com/Expensify/App/issues/61450

**Pull request merged**